### PR TITLE
Add docs about new git workflow

### PR DESCRIPTION
**Pull request by ncoden**
Changes:
- Add docs about the branches of the new git workflow in `CONTRIBUTING.md`
- Update GitHub pull request template

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,6 +22,13 @@ All new features and bug fixes should be submitted as pull requests, so the comm
 
 When you submit a pull request, @mention a few people you’d like to help you review it. Once those people have signed off on it, the pull request can be merged! Core Team members will handle the merge itself.
 
+## Git Worflow
```
Comment (kball) [inline CONTRIBUTING.md]: Typo -  Workflow not 'Worflow'

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -22,6 +22,13 @@ All new features and bug fixes should be submitted as pull requests, so the comm
...
+
+Foundation uses a git workflow close to [GitFlow](http://nvie.com/posts/a-successful-git-branching-model/). The workflow relies on three branches:
+- **`master`**: The stable branch. It only contains the latest stable version released. Pull requests for docs improvements can be opened on it.
+- **`develop`**: The developing branch used to prepare the next version. It always contains all the new features and bug fixes previously made, and constitutes the more up-to-date version of Foundation. Almost every pull request should be open on this branch. When a new version is released, it is merged on `master`, a support branch is created, and `develop` now target the next version.
```
Comment (kball) [inline CONTRIBUTING.md]: Should we highlight that these are major/minor versions? To make clear that 'version' doesn't refer to patch versions...

Also, English feedback: "more up-to-date" => "most up-to-date". "now target the next" => "now targets the next"

**Review by kball — CHANGES_REQUESTED** [review]
One final piece of feedback, lets note that this workflow has been adopted as of 6.3, so folks don't ask about a support/6.2 branch etc

File: CONTRIBUTING.md (hunk 2 above)
Comment (ncoden) [inline CONTRIBUTING.md]: You are right.

**Comment by kball** [comment]
This makes sense to me, can I get another review from someone on @zurb/yetinauts to make sure this seems clear to you and communicates our new process?

**Comment by IamManchanda** [comment]
so now new features will go to develop ..... ?

**Comment by ncoden** [comment]
@IamManchanda Yes.

@zurb/yetinauts Someone to review it ?

**Comment by colin-marshall** [comment]
@ncoden @kball looks good to me!

**Pull request merged**